types/react-fontawesome/index.d.ts

```diff
@@ -11,12 +11,16 @@ import * as React from 'react';
 
 export = FontAwesome;
 
+interface Intermediate extends React.HTMLAttributes<HTMLElement> {
+    size?: any;
```

Comment:
`React.HTMLAttributes<T>` used to define `size?: number` in v15. The intermediate is to get around that for anyone using an older version. Extending `React.AllHTMLAttributes<T>` might be more appropriate, but that hasn't always been provided for v15.

Reply:
That is strange, in my knowledge, there must have been an error in the typedefs then, since:

```
interface A {
    size: number;
}

interface B extends A {
    size?: 'lg' | '2x' | '3x' | '4x' | '5x';
}
```

will say something like: `types of property 'size' are incompatible.`

I may not understand something, but I don't see how your proposed solution helps with the problem you describe. A quick test which illustrates your problem (I think):

```
interface C {
    size?: number;
}

interface A extends C {
    size?: any;
}

interface B extends A {
    size?: 'lg' | '2x' | '3x' | '4x' | '5x';
}

const a: B = { size: 5 }
```

const a is underlined with an error: `Type 'number' is not assignable to type '"lg" | "2x" | "3x" | "4x" | "5x"'`, which seems reasonable to me.

Reply:
>But @types/react is on v16 now -- if people are still using the old @types/react, they're probably not going to use the latest version of this package anyway.

If we want to assume @types/react@16, I'd rather extend `AllHTMLAttributes` if people are using other tags which still presents the same issue with size. I'll make that push here shortly.

> That is strange, in my knowledge, there must have been an error in the typedefs then, since:

Correct, you'll get that error (with your example), which is a problem. `size` *should* be `'lg' | '2x' | '3x' | '4x' | '5x'`. To allow for size to be redefined to `'lg' | '2x' | '3x' | '4x' | '5x'`, the intermediate interface is needed which redefines `size?: any` so size can then be defined to `size?: 'lg' | '2x' | '3x' | '4x' | '5x'`. It's a 'hack' to get around that problem. Remember, previously this was a type union, not an interface extension.

>const a is underlined with an error: Type 'number' is not assignable to type '"lg" | "2x" | "3x" | "4x" | "5x"'

Exactly, which is what should happen. You want size to now be the possible sizes, not a number. If you instead do `const foo: B = { size: 'lg' }`, you're fine. react-fontawesome hijacks the `size` attribute. [See source](https://github.com/danawoodman/react-fontawesome/blob/master/src/index.js#L41).